onnxmltools/convert/keras/operator_converters/Bidirectional.py

```diff
@@ -0,0 +1,251 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import numpy as np
+from distutils.version import StrictVersion
+from keras.layers import Bidirectional, LSTM
+from ...common._apply_operation import apply_transpose, apply_split, apply_reshape
+from ...common._registration import register_converter
+from ....proto import onnx_proto
+from .common import extract_recurrent_activation
+
+
+def convert_bidirectional(scope, operator, container):
+    # Extract basic information and create aliases for some fields
+    op = operator.raw_operator
+    forward_layer = op.forward_layer
+    backward_layer = op.backward_layer
+    input_size = op.input_shape[-1]
+    seq_length = op.input_shape[-2]
+    hidden_size = forward_layer.units
+    output_seq = forward_layer.return_sequences
+    output_state = forward_layer.return_state
+    if output_state:
+        raise ValueError('Keras Bidirectional cannot return hidden and cell states')
+    if not isinstance(forward_layer, LSTM):
+        raise TypeError('The bidirectional module only works with LSTM in Keras but we got %s' % type(forward_layer))
+
+    # Extract the forward transformation matrix used to adjust input features
+    W_x = np.empty(shape=(4, hidden_size, input_size))
+    K_W_x = forward_layer.get_weights()[0].T
+    W_x[0:] = K_W_x[0 * hidden_size:][:hidden_size]
+    W_x[1:] = K_W_x[3 * hidden_size:][:hidden_size]
+    W_x[2:] = K_W_x[1 * hidden_size:][:hidden_size]
+    W_x[3:] = K_W_x[2 * hidden_size:][:hidden_size]
+
+    # Extract the forward transformation matrix used to adjust hidden state
+    W_h = np.empty(shape=(4, hidden_size, hidden_size))
+    K_W_h = forward_layer.get_weights()[1].T
+    W_h[0:] = K_W_h[0 * hidden_size:][:hidden_size]
+    W_h[1:] = K_W_h[3 * hidden_size:][:hidden_size]
+    W_h[2:] = K_W_h[1 * hidden_size:][:hidden_size]
+    W_h[3:] = K_W_h[2 * hidden_size:][:hidden_size]
+
+    # Bias vectors of forward layer
+    b = None
+    if forward_layer.use_bias:
+        b = np.zeros(shape=(8, hidden_size))
+        keras_b = forward_layer.get_weights()[2]
+        b[0:] = keras_b[0 * hidden_size:][:hidden_size]
+        b[1:] = keras_b[3 * hidden_size:][:hidden_size]
```

Comment:
I almost forget the order coming from, can you add some comments here?

Reply:
Sure. Will add it.